stock_picking_mto/models/stock_picking.py

```diff
@@ -0,0 +1,40 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def _get_selection_procure_method(self):
+        return self.env['stock.move'].fields_get(
+            allfields=['procure_method'])['procure_method']['selection']
+
+    def _compute_procure_method(self):
```

Comment:
@chienandalu Even in pure computed method, add a depends on move_lines.procure_method to fetch the recordset before entering the loop.